sympy/assumptions/handlers/order.py

```diff
@@ -66,32 +66,88 @@ def Add(expr, assumptions):
         if r is not True:
             return r
 
-        nonpos = 0
-        for arg in expr.args:
-            if ask(Q.negative(arg), assumptions) is not True:
-                if ask(Q.positive(arg), assumptions) is False:
-                    nonpos += 1
-                else:
-                    break
-        else:
-            if nonpos < len(expr.args):
-                return True
+
+        args = [arg for arg in expr.args if not ask(Q.zero(arg), assumptions)]
+        if not args:
+            return False
+
+        inf = neg_inf = pos_inf = neg = npos = nneg = False
+
+        for arg in args:
+            if pos_inf and neg_inf:
+                return None
+
+            _inf = ask(Q.infinite(arg), assumptions)
+            if _inf:
+                inf = True
+
+            n = ask(Q.negative(arg), assumptions)
+            if n:
+                if _inf is not False:
+                    neg_inf = True
+                neg = True
+                continue
+
+            np = ask(Q.nonpositive(arg), assumptions)
+            if np:
+                if _inf is True:
+                    neg_inf = True
+                npos = True
+                continue
+
+            nn = ask(Q.nonnegative(arg), assumptions)
+            if nn:
+                if _inf or (_inf is None and nn is False):
```

Comment:
This second condition is unreachable, `nn` cannot be `False` inside `if nn`
```suggestion
                if _inf:
```

Reply:
Done.

Reply:
I have undone the change here as Travis CI test is failing without "or (_inf is None and nn is False):"

Reply:
I can't see how that can possibly be true. If this line runs, then `nn` must be truthy. Therefore, we can simplify as:
```python
_inf or (_inf is None and nn is False)
_inf or (_inf is None and False)
_inf or (False)
_inf
```
Unless I've made a mistake above.